src/test/java/org/owasp/esapi/reference/HTTPUtilitiesTest.java

```diff
@@ -264,6 +265,115 @@ public void testGetFileUploads() throws Exception {
+     * and if you look through the exception stack trace, you can see the
+     * 'Caused by' reason is:
+     *      Caused by: org.apache.commons.fileupload.FileCountLimitExceededException: attachment
+     *          at org.apache.commons.fileupload.FileUploadBase.parseRequest(FileUploadBase.java:367)
+     *          at org.apache.commons.fileupload.servlet.ServletFileUpload.parseRequest(ServletFileUpload.java:113)
+     *          at org.owasp.esapi.reference.DefaultHTTPUtilities.getFileUploads(DefaultHTTPUtilities.java:628)
+     *          ... 23 more
+     * which is as it should be.
+     *
+     */
+    public void testGetFileUploadsTooManyFiles() throws Exception {
+        File home = null;
+
+        System.out.println("testGetFileUploadsTooManyFiles");
+
+        try
+        {
+            home = FileTestUtils.createTmpDirectory(CLASS_NAME);
+            String content =    "Content-Type: multipart/form-data; boundary=ridiculous\r\n\r\n\r\n" +
+                                "--ridiculous\r\n" +
+                                "Content-Disposition: form-data; name=\"full-name\"\r\n\r\n" +
+                                "kevin w wall\r\n" +
+                                "--ridiculous\r\n" +
+                                "Content-Disposition: form-data; name=\"file1\"; filename=\"aaa.txt\"\r\n" +
+                                "Content-Type: text/plain\r\n\r\n" +
+                                "AAA\r\n\r\n" +
+                                "--ridiculous\r\n" +
+                                "Content-Disposition: form-data; name=\"file2\"; filename=\"bbb.txt\"\r\n" +
+                                "Content-Type: text/plain\r\n\r\n" +
+                                "BBB\r\n\r\n" +
+                                "--ridiculous\r\n" +
+                                "Content-Disposition: form-data; name=\"file3\"; filename=\"ccc.txt\"\r\n" +
+                                "Content-Type: text/plain\r\n\r\n" +
+                                "CCC\r\n\r\n" +
+                                "--ridiculous--\r\n\r\n";
+
+            MockHttpServletResponse response = new MockHttpServletResponse();
+            MockHttpServletRequest request1 = new MockHttpServletRequest("/test", content.getBytes(response.getCharacterEncoding()));
+            ESAPI.httpUtilities().setCurrentHTTP(request1, response);
+
+            MockHttpServletRequest request2 = new MockHttpServletRequest("/test", content.getBytes(response.getCharacterEncoding()));
+            request2.setContentType( "multipart/form-data; boundary=ridiculous");
+            ESAPI.httpUtilities().setCurrentHTTP(request2, response);
+            List<File> response2 = new ArrayList<>();
+            boolean caughtExpectedException = false;
+            try {
+                response2 = ESAPI.httpUtilities().getFileUploads(request2, home);
+            } catch( ValidationUploadException vuex ) {
+                caughtExpectedException = true;
```

Comment:
There are built-in ways to assert that a test fails as expected using content like the ExpectedException rule (junit4) or using `@Test (expected=ValidationUploadException.class)`

Just fyi points that achieve the same general goal.

There are possible optimizations for this test, but I believe that overall it achieves the desired effect.

Reply:
Good point. I will fix that. I probably need to import the appropriate annotation though as well too. And I will also add @Ignore annotation instead of commenting everything out from that one test like I did which is just ugly.

Reply:
The @Ignore annotation didn't work, probably because we are still using the JUnit 3 stype TestSuite test setup, so I just left the testGetFileUploadsUnauthenticatedUser commented out.

Reply:
My apologies for not returning to this one in time, I can add refactoring this test to the percent codec update for UTF-8 on my TODO.  This one will probably be done sooner.